app/controllers/nsm/imports_controller.rb

```diff
@@ -23,22 +23,20 @@ def create
       if @form_object.valid?
         initialize_application do |claim|
           @validation_errors = validate
-
           if @validation_errors.empty?
-            hash = claim_hash
-
-            # TODO: CRM457-2473: Refactor this to handle versioning better
-            Nsm::Importers::Xml.const_get("v#{xml_file.version.to_i}".capitalize)::Importer.new(claim, hash).call
-
-            redirect_to edit_nsm_steps_claim_type_path(claim.id), flash: { success: build_message(claim) } and return
+            process_successful_import(claim)
+            return # Exit the action after redirect
           else
-            errors_file_path.write(@validation_errors.to_json)
-            @form_object.errors.add(:file_upload, :validation_errors)
+            handle_validation_errors
           end
         end
       end
 
       render :new
+    rescue ActionController::ParameterMissing
```

Comment:
This isn't needed, it's already handled as part of the new `before_action`

Reply:
Thanks, It was before I merged the main. updating now.